client/src/socket.js

```diff
@@ -0,0 +1,25 @@
+import io from 'socket.io-client';
+
+// Connect to socket.io server
+export default function () {
+    const socket = io.connect('http://localhost:5000/socket.io');
+
+    //TODO update message states
+    socket.on('refresh message', function (err, conversationId) {
```

Comment:
This needs to be wrapped in a function like `registerListener(callback) {...}` and included in the `return {...}` so I can pass in the state setting methods from the React component